data/maps/SP_Scenarios/Along_The_River.wmf/scripting/objectives.lua

```diff
@@ -0,0 +1,93 @@
+-- =======================================================================
+--                             Along The River
+-- =======================================================================
+
+obj_defeat_red = {
+   name = "defeat_red",
+   title = _"Defeat Törver",
+   number = 1,
+   body = objective_text(_"Defeat the armies of Törver.",
+      li_image("map:banner_plr_02.png", _[[Törver is the enemy in the north, who is fighting under the red banner.]])
+   ),
+}
+
+obj_build_port = {
+   name = "build_port",
+   title = _"Build a port",
+   number = 1,
+   body = objective_text(_"Build a port in the land of Törver.",
+      li_image("tribes/buildings/warehouses/barbarians/port/menu.png", _[[We should be ready to send an expedition as soon as we found out how to build ships.]])
+   ),
+}
+
+obj_conquer_pirate_sentry = {
+   name = "conquer_pirate_sentry",
+   title = _"Conquer pirate sentry",
+   number = 1,
+   body = objective_text(_"Conquer the pirate sentry near the brook to the east.",
+      li_image("tribes/buildings/militarysites/barbarians/sentry/menu.png", _[[Hopefully we will find out what happened to the gold in this area.]])
+   ),
+}
+
+obj_build_warehouse = {
+   name = "build_warehouse",
+   title = _"Build warehouse",
+   number = 1,
+   body = objective_text(_"Build a warehouse close to the pirate sentry near the brook.",
+      li_image("tribes/buildings/warehouses/barbarians/warehouse/menu.png", _[[We need storage for the gold to be mined.]])
```

Comment:
This sentence is unclear – do we need storage in order to mine gold, or storage for the gold we intend/hope/plan to mine

Reply:
technically the warehouse is needed to store the shovels for removing the swamp. from the story line it is intended to store the gold that is going to be m ined from the surrounding hills